.github/workflows/full-check.yml

```diff
@@ -4,23 +4,46 @@ on: [push, pull_request]
 
 jobs:
   build:
+    strategy:
+      fail-fast: false
+      matrix:
+        include:
+          - cc: gcc-13
+            clang_major_version: null
+            runs-on: ubuntu-22.04
+          - cc: clang-18
+            clang_major_version: 18
+            runs-on: ubuntu-22.04
 
-    runs-on: ubuntu-latest
+    runs-on: ${{ matrix.runs-on }}
 
     steps:
     - uses: actions/checkout@v4
+    - name: clang_repo
+      if: "${{ contains(matrix.cc, 'clang') }}"
+      run: |
+           set -x
+           source /etc/os-release
+           wget -O - https://apt.llvm.org/llvm-snapshot.gpg.key | sudo apt-key add -
+           sudo add-apt-repository "deb http://apt.llvm.org/${UBUNTU_CODENAME}/ llvm-toolchain-${UBUNTU_CODENAME}-${{ matrix.clang_major_version }} main"
```

Comment:
Couldn't we use the clang version already present in Ubuntu's repositories?

Reply:
@davidpolverari it's always ages behind. Upstream is in the middle of Clang 19 right now and I'm only not adding that because their renames always break CI at some point when you need it the least. Ubuntu 22.04 — latest in GitHub Actions is at Clang 16. I wanted seriously modern, in particular, because compilers get stricter by the day.

Reply:
@davidpolverari PS: correcting that to Clang 15 based on https://github.com/actions/runner-images/blob/main/images/ubuntu/Ubuntu2204-Readme.md .